cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/pcluster_slurm_config_generator.py

```diff
@@ -307,6 +319,82 @@ def _realmemory(compute_resource, realmemory_to_ec2memory_ratio) -> int:
     return realmemory
 
 
+def _check_leading_slash(uri: str) -> bool:
+    if uri.startswith("/"):
+        error_msg = (
+            f"Invalid URI specified. Please remove any leading / at the beginning of the provided URI ('{uri}')."
+        )
+        log.critical(error_msg)
+        raise CriticalError(error_msg)
+    return True
+
+
+def _check_scheme(uri: str, uri_parse: ParseResult) -> bool:
+    try:
+        scheme = uri_parse.scheme
+    except ValueError as e:
+        error_msg = (f"Failure to parse uri with error '{str(e)}'. Please review the provided URI ('{uri}')",)
+        log.critical(error_msg)
+        raise CriticalError(error_msg)
+    if scheme:
+        error_msg = (f"Invalid URI specified. Please do not provide a scheme ('{scheme}://')",)
+        log.critical(error_msg)
+        raise CriticalError(error_msg)
+    return True
+
+
+def _parse_netloc(uri: str, uri_parse: ParseResult, attr: str) -> str:
+    try:
+        netloc = uri_parse.netloc
+    except ValueError as e:
+        error_msg = f"Failure to parse uri with error '{str(e)}'. Please review the provided URI ('{uri}')"
+        log.critical(error_msg)
+        raise CriticalError(error_msg)
+    if not netloc:
+        error_msg = f"Invalid URI specified. Please review the provided URI ('{uri}')"
+        log.critical(error_msg)
+        raise CriticalError(error_msg)
+    if attr == "host":
+        ret = uri_parse.hostname
+    elif attr == "port":
+        ret = uri_parse.port
+        # Provide default MySQL port if port is not explicitely set
+        if not ret:
+            ret = "3306"
+    return ret
+
+
+def _parse_uri(uri, attr) -> str:
+    """Get a host from a URI/URL using urlparse."""
+    # First, throw error if the URI starts with a "/" (to prevent issues with the
+    # manipulation below
+    if not _check_leading_slash(uri):
+        return None
```

Comment:
This and the if below with _check_scheme are a bit confusing because of the following:
* by reading this function it gives the impression you are returning None in case the check fails. And returning None  would produce an invalid template.
* However what it actually happens is that the function raises an exception and the `return None` line is just dead code.

I'd suggest to make this more consistent/readable. Either have the checker function always return True/False and have the caller react on that or transform this in assertions and simply call them like `_check_leading_slash(uri)` without and if/return logic.

Reply:
Also, more of a general question, can't we assume that this checks are done by the CLI that validates the config and do not need to be repeated here?

Reply:
About your first comment, I had implemented the same structure from the CLI side and left the same `if not ... return` structure, which here I probably didn't need. Thanks.

For your second comment, we have a "backdoor" when users decide to disable the validation on the CLI side. In this case should we allow the cookbook to blindly accept whatever the users provide as input?

Reply:
OK to remove it from the cookbook side. To be done.

Reply:
Done.